registry.go

```diff
@@ -104,6 +105,11 @@ type Layer interface {
 
 	// CreatedAt returns the time this layer was created.
 	CreatedAt() time.Time
+
+	// ServeHTTP allows a layer to serve itself, whether by providing
+	// a redirect directly to the content, or by serving the content
+	// itself
+	ServeHTTP(w http.ResponseWriter, r *http.Request)
```

Comment:
@endophage Sounds reasonable. Although, a caller would not be able to detect the error condition and take another action. Deferring this will have minimal impact.